packages/react-aria/src/hooks/useAriaButton.ts

```diff
@@ -0,0 +1,94 @@
+import * as React from 'react';
+import { ObjectShorthandProps, useEventCallback } from '@fluentui/react-utilities';
+
+export function useARIAButton(shorthand: ObjectShorthandProps<React.ButtonHTMLAttributes<HTMLElement>>) {
+  if (!shorthand.hasOwnProperty('as')) {
+    shorthand.as = 'button';
+  }
+  if (!shorthand.hasOwnProperty('children')) {
+    shorthand.children = null;
+  }
+
+  const { onClick, onKeyDown, onKeyUp, as, disabled } = shorthand;
+
+  const onClickHandler = useEventCallback((ev: React.MouseEvent<HTMLElement>) => {
+    if (disabled) {
+      ev.preventDefault();
+    } else {
+      onClick?.(ev);
+    }
+  });
+
+  const onKeyDownHandler = useEventCallback((ev: React.KeyboardEvent<HTMLElement>) => {
+    onKeyDown?.(ev);
+    if (ev.isDefaultPrevented()) {
+      return;
+    }
+    if (disabled && (ev.key === 'Enter' || ev.key === ' ')) {
```

Comment:
question is if there is no other package in REAct Examples that migh use that

Reply:
if you use that utility at the top, you should only need to prevent default once in the key handlers

Reply:
I avoided using that since there's a dependency on `@fluentui/keyboard-key`, which I wanted to avoid